### PR TITLE
Use target-features to etablish hvx single/double modes

### PR DESCRIPTION
**Pull request by ronlieb**

### CODE REVIEW

File: src/CodeGen_Hexagon.cpp [hunk 1]
```diff
@@ -51,14 +51,20 @@ std::unique_ptr<llvm::Module> CodeGen_Hexagon::compile(const Module &module) {
...
     if (module.target().has_feature(Halide::Target::HVX_128)) {
-        char *s = strdup("HALIDE_LLVM_INTERNAL=-enable-hexagon-hvx-double");
-        ::putenv(s);
-        cl::ParseEnvironmentOptions("halide-hvx-be", "HALIDE_LLVM_INTERNAL",
-                                    "Halide HVX internal options\n");
+        hvx_setting = "+hvx,+hvx-double";
         if (module.target().has_feature(Halide::Target::HVX_64))
                 internal_error << "Both HVX_64 and HVX_128 set at same time\n";
     }
+    for (llvm::Module::iterator iter = llvm_module->begin();
```
Comment (dsharletg) [inline src/CodeGen_Hexagon.cpp]: We'd prefer:

    for (llvm::Function &fn : llvm_module->functions()) {
        ...
    }

**Comment by dsharletg** [comment]
Is this supported on the LLVM we have access to?

**Comment by ronlieb** [comment]
i beleive it is supported in the upstream llvm.org.
i do see the same .td file entry as in our local repo.

Regarding, the coding suggestion, i will make and test the change today and post a follow on patch.

**Comment by dsharletg** [comment]
Thanks for fixing this! I'll merge it.

Btw, it doesn't notify us if you push a commit to the branch, so after you fix things and push commits to a PR branch, you should add a comment saying as much :)

**Pull request merged**